client/src/main/scala/com/redi/scalademo/application/ClientModule.scala

```diff
@@ -0,0 +1,13 @@
+package com.redi.scalademo.application
+
+import com.redi.scalademo.business.NumericStringValidator
+import com.redi.scalademo.infrastructure.Client
+import com.redi.scalademo.presentation.FormListener
+
+private[application] object ClientModule {
+
+  lazy val numericStringValidator = new NumericStringValidator
+  lazy val client = new Client
+  lazy val formListener = new FormListener(client, numericStringValidator)
```

Comment:
I think there is a value in demoing DI framework and how the DI container (or module) is shareable between JS and JVM. Why don't you use [MacWire](https://github.com/adamw/macwire)?